File: app/src/main/java/com/killain/organizer/packages/interactors/DateHelper.java
package com.killain.organizer.packages.interactors;

import com.killain.organizer.packages.enums.FormatDateOutput;

import org.threeten.bp.DayOfWeek;
import org.threeten.bp.LocalDate;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;

public class DateHelper {

    private String day, month, result, time, date, task_date, minute, hour;
    private int int_hour, int_minute;
    private LocalDate localDate;
    private char zeroChar = 0;

    public DateHelper() {
        localDate = LocalDate.now();
    }

    private String formatMonth(int month) {
        String convertedMonth = null;
        switch (month) {
            case 1:
                convertedMonth = "Jan";
                break;

            case 2:
                convertedMonth = "Feb";
                break;

            case 3:
                convertedMonth = "Mar";
                break;

            case 4:
                convertedMonth = "Apr";
                break;

            case  5:
                convertedMonth = "May";
                break;

            case 6:
                convertedMonth = "Jun";
                break;

            case 7:
                convertedMonth = "Jul";
                break;

            case 8:
                convertedMonth = "Aug";
                break;

            case 9:
                convertedMonth = "Sep";
                break;

            case 10:
                convertedMonth = "Oct";
                break;

            case 11:
                convertedMonth = "Nov";
                break;

            case 12:
                convertedMonth = "Dec";
                break;
        }
        return convertedMonth;
    }

    public String localDateToString(LocalDate localDate, FormatDateOutput formatDateOutput) {
        LocalDate compare = LocalDate.now();

        if (compare.getYear() == localDate.getYear() &&
                compare.getMonthValue() == localDate.getMonthValue() &&
                compare.getDayOfMonth() == localDate.getDayOfMonth() &&
                formatDateOutput == FormatDateOutput.FORMAT_DATE_OUTPUT) {
            result = "Today";
            return result;
        } else {
            String dow = convertLocalDateDayOfWeek(localDate.getDayOfWeek());
            String dayOfMonth = Integer.toString(localDate.getDayOfMonth());
            String month = formatMonth(localDate.getMonthValue());
            String year = Integer.toString(localDate.getYear());
            result = dow + ", " + month + " " + dayOfMonth + ", " + year;
            return result;
        }
    }

    public String localDateToTaskString(LocalDate localDate) {
            String year = Integer.toString(localDate.getYear());
            String month = Integer.toString(localDate.getMonthValue());
            String day = Integer.toString(localDate.getDayOfMonth());
            task_date = day + "/" + month + "/" + year;
            return task_date;
    }

    public void setDate(int year, int month, int day) {
        localDate = LocalDate.of(year, month, day);
    }

    public void setTime(int hour, int minute) {
        this.hour = Integer.toString(hour);
        this.minute = Integer.toString(minute);

        if (minute < 9) {
            this.minute = "0" + Integer.toString(minute);
        }

        if (hour < 9) {
            this.hour = "0" + Integer.toString(hour);
        }

        time = this.hour + ":" + this.minute;
    }

    public String getFullDateWithTime() {
        date = localDateToString(localDate, FormatDateOutput.DEFAULT);
        result = date + " " + time;
        return result;
    }

    public int getYear() {
        return localDate.getYear();
    }

    public int getMonth() {
        return localDate.getMonthValue();
    }

    public int getDayOfMonth() {
        return localDate.getDayOfMonth();
    }

    public String getConvertedDateFromLong(long long_date) {
        localDate = LocalDate.ofEpochDay(long_date);
        return localDateToString(localDate, FormatDateOutput.FORMAT_DATE_OUTPUT);
    }

    private String convertLocalDateDayOfWeek(DayOfWeek dayOfWeek) {

        String converted_dow = null;

        switch (dayOfWeek) {

            case MONDAY:
                converted_dow = "Mon";
                break;

            case TUESDAY:
                converted_dow = "Tue";
                break;

            case WEDNESDAY:
                converted_dow = "Wed";
                break;

            case THURSDAY:
                converted_dow =  "Thu";
                break;

            case FRIDAY:
                converted_dow = "Fri";
                break;

            case SATURDAY:
                converted_dow = "Sat";
                break;
            case SUNDAY:
                converted_dow = "Sun";
                break;

                default: break;
        }
        return converted_dow;
    }

    public void setLocalDate(LocalDate localDate) {
        this.localDate = localDate;
    }

    public long getLong() {
        return localDate.toEpochDay();
    }
}

File: app/src/main/java/com/killain/organizer/packages/interactors/DateHelper.kt
package com.killain.organizer.packages.interactors

import com.killain.organizer.packages.enums.FormatDateOutput

import org.threeten.bp.DayOfWeek
import org.threeten.bp.LocalDate

import java.text.SimpleDateFormat
import java.util.Calendar
import java.util.Date

class DateHelper {

    private val day: String? = null
    private val month: String? = null
    private var result: String? = null
    private var time: String? = null
    private var date: String? = null
    private var task_date: String? = null
    private var minute: String? = null
    private var hour: String? = null
    private val int_hour: Int = 0
    private val int_minute: Int = 0
    private var localDate: LocalDate? = null
    private val zeroChar: Char = 0.toChar()

    val fullDateWithTime: String
        get() {
            date = localDateToString(localDate, FormatDateOutput.DEFAULT)
            result = "$date $time"
            return result as String
        }

    val year: Int?
        get() = localDate?.year

    val dayOfMonth: Int?
        get() = localDate?.dayOfMonth

    val long: Long?
        get() = localDate?.toEpochDay()

    init {
        localDate = LocalDate.now()
    }

    private fun formatMonth(month: Int?): String? {
        var convertedMonth: String? = null
        when (month) {
            1 -> convertedMonth = "Jan"

            2 -> convertedMonth = "Feb"

            3 -> convertedMonth = "Mar"

            4 -> convertedMonth = "Apr"

            5 -> convertedMonth = "May"

            6 -> convertedMonth = "Jun"

            7 -> convertedMonth = "Jul"

            8 -> convertedMonth = "Aug"

            9 -> convertedMonth = "Sep"

            10 -> convertedMonth = "Oct"

            11 -> convertedMonth = "Nov"

            12 -> convertedMonth = "Dec"
        }
        return convertedMonth
    }

    fun localDateToString(localDate: LocalDate?, formatDateOutput: FormatDateOutput): String {
        val compare = LocalDate.now()

        return if (compare.year == localDate?.year &&
                compare.monthValue == localDate.monthValue &&
                compare.dayOfMonth == localDate.dayOfMonth &&
                formatDateOutput === FormatDateOutput.FORMAT_DATE_OUTPUT) {
            result = "Today"
            result as String
        } else {
            val dow = convertLocalDateDayOfWeek(localDate?.dayOfWeek)
            val dayOfMonth = Integer.toString(localDate!!.dayOfMonth)
            val month = formatMonth(localDate.monthValue)
            val year = Integer.toString(localDate.year)
            result = "$dow, $month $dayOfMonth, $year"
            result as String
        }
    }

    fun localDateToTaskString(localDate: LocalDate): String {
        val year = Integer.toString(localDate.year)
        val month = Integer.toString(localDate.monthValue)
        val day = Integer.toString(localDate.dayOfMonth)
        task_date = "$day/$month/$year"
        return task_date as String
    }

    fun setDate(year: Int, month: Int, day: Int) {
        localDate = LocalDate.of(year, month, day)
    }

    fun setTime(hour: Int, minute: Int) {
        this.hour = Integer.toString(hour)
        this.minute = Integer.toString(minute)

        if (minute < 9) {
            this.minute = "0" + Integer.toString(minute)
        }

        if (hour < 9) {
            this.hour = "0" + Integer.toString(hour)
        }

        time = this.hour + ":" + this.minute
    }

    fun getMonth(): Int? {
        return localDate!!.monthValue
    }

    fun getConvertedDateFromLong(long_date: Long?): String {
        if (long_date != null) {
            localDate = LocalDate.ofEpochDay(long_date)
        }
        return localDateToString(localDate, FormatDateOutput.FORMAT_DATE_OUTPUT)
    }

    private fun convertLocalDateDayOfWeek(dayOfWeek: DayOfWeek?): String? {

        var converted_dow: String? = null

        when (dayOfWeek) {

            DayOfWeek.MONDAY -> converted_dow = "Mon"

            DayOfWeek.TUESDAY -> converted_dow = "Tue"

            DayOfWeek.WEDNESDAY -> converted_dow = "Wed"

            DayOfWeek.THURSDAY -> converted_dow = "Thu"

            DayOfWeek.FRIDAY -> converted_dow = "Fri"

            DayOfWeek.SATURDAY -> converted_dow = "Sat"
            DayOfWeek.SUNDAY -> converted_dow = "Sun"

            else -> {
            }
        }
        return converted_dow
    }

    fun setLocalDate(localDate: LocalDate) {
        this.localDate = localDate
    }
}
